src/circuit/gadget/ecc/chip/mul.rs

```diff
@@ -0,0 +1,427 @@
+        &self,
+        scalar: &CellValue<pallas::Base>,
+        base: &EccPoint,
+        offset: usize,
+        region: &mut Region<'_, pallas::Base>,
+    ) -> Result<EccPoint, Error> {
+        // Decompose the scalar bitwise (big-endian bit order).
+        let bits = decompose_for_scalar_mul(scalar.value());
+        let bits_incomplete_hi = &bits[INCOMPLETE_HI_RANGE];
+        let bits_incomplete_lo = &bits[INCOMPLETE_LO_RANGE];
+        let lsb = bits[pallas::Scalar::NUM_BITS as usize - 1];
+
+        // Initialize the accumulator `acc = [2]base`
+        let acc = self
+            .add_config
+            .assign_region(&base, &base, offset, region)?;
+
+        // Increase the offset by 1 after complete addition.
+        let offset = offset + 1;
+
+        // Initialize the running sum for scalar decomposition to zero
+        let z_init = {
+            // Constrain the initialization of `z` to equal zero.
+            self.q_init_z.enable(region, offset)?;
+
+            let z_val = pallas::Base::zero();
+            let z_cell =
+                region.assign_advice(|| "initial z", self.hi_config.z, offset, || Ok(z_val))?;
+
+            Z(CellValue::new(z_cell, Some(z_val)))
+        };
+
+        // Increase the offset by 1 after initializing `z`.
+        let offset = offset + 1;
+
+        // Double-and-add (incomplete addition) for the `hi` half of the scalar decomposition
+        let (x_a, y_a, zs_incomplete_hi) = self.hi_config.double_and_add(
+            region,
+            offset,
+            &base,
+            bits_incomplete_hi,
+            (X(acc.x), Y(acc.y.value()), z_init),
+        )?;
+
+        // Double-and-add (incomplete addition) for the `lo` half of the scalar decomposition
+        let z = &zs_incomplete_hi[zs_incomplete_hi.len() - 1];
+        let (x_a, y_a, zs_incomplete_lo) = self.lo_config.double_and_add(
+            region,
+            offset,
+            &base,
+            bits_incomplete_lo,
+            (x_a, y_a, *z),
+        )?;
+
+        // Move from incomplete addition to complete addition.
+        // Inside incomplete::double_and_add, the offset was increase once after initialization
+        // of the running sum.
+        // Then, the final assignment of double-and-add was made on row + offset + 1.
+        // Outside of incomplete addition, we must account for these offset increases by adding
+        // 2 to the incomplete addition length.
+        let offset = offset + INCOMPLETE_LO_RANGE.len() + 2;
+
+        // Complete addition
+        let (acc, zs_complete) = {
+            let z = &zs_incomplete_lo[zs_incomplete_lo.len() - 1];
+            let complete_config: complete::Config = self.into();
+            // Bits used in complete addition. k_{3} to k_{1} inclusive
+            // The LSB k_{0} is handled separately.
+            let bits_complete = &bits[COMPLETE_RANGE];
+            complete_config.assign_region(region, offset, bits_complete, base, x_a, y_a, *z)?
```

Comment:
This is assigning `y_a` to a cell internally, but isn't connecting it to the incomplete addition output (because as above, `double_and_add` only returns a value). Once `double_and_add` returns a `CellValue`, we can just pass that in here and copy it into the complete addition.